evaluation_plans/osps/quality/steps.go

```diff
@@ -124,18 +124,24 @@ func statusChecksAreRequiredByBranchProtection(payloadData interface{}, _ map[st
 	return layer4.Passed, "No status checks were run that are not required by branch protection"
 }
 
-// TODO: after 3 layers of depth, make additional API calls if a tree is found.
-// TODO: Examine more than just the file name.
 func noBinariesInRepo(payloadData interface{}, _ map[string]*layer4.Change) (result layer4.Result, message string) {
 	data, message := reusable_steps.VerifyPayload(payloadData)
 	if message != "" {
 		return layer4.Unknown, message
 	}
 
-	if len(data.SuspectedBinaries) == 0 {
-		return layer4.Passed, "No binaries were found in the repository (Note: this check only examines file names at this time)"
+	// TODO: This only checks the top 3 levels of the repository tree
+	// for common binary file extensions and it fails on very large repositories.
+	suspectedBinaries, err := data.GetSuspectedBinaries()
+	if err != nil {
+		data.Config.Logger.Trace(fmt.Sprintf("unexpected response while checking for binaries: %s", err.Error()))
+		return layer4.Unknown, "Error while scanning repository for binaries, potentially due to repo size. See logs for details."
```

Comment:
I'm a fan of the fail fast method. I'm asking other clarifying questions privately but this is good for me.